gossip/src/gossip_service.rs

```diff
@@ -143,10 +143,7 @@ pub fn discover(
     Vec<ContactInfo>, // all gossip peers
     Vec<ContactInfo>, // tvu peers (validators)
 )> {
-    let keypair = {
-        #[allow(clippy::redundant_closure)]
-        keypair.unwrap_or_else(|| Keypair::new())
```

Comment:
Maybe I'm missing something? Doesn't seem like we need this.

Reply:
Looks like it, yeah. I got a clippy complaint about this at one point but glad it's gone